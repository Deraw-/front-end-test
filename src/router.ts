import Vue from 'vue';
import Router from 'vue-router';
import Meta from 'vue-meta';

import i18n, { loadLanguageAsync } from '@/i18n';

// Directly import Anomalies view for faster rendering of first page
import Home from '@/views/Home.vue';

Vue.use(Router);
Vue.use(Meta);

// See https://router.vuejs.org/fr/essentials/navigation.html for help
const router = new Router({
	mode: 'history', // Use the History API, and remove the hash from the URL
	routes: [
		{
			path: '/',
			name: 'home',
			component: Home
		},
		{
			path: '/cart',
			name: 'cart',
			component: () => import(/* webpackChunkName: "cart" */ './views/Cart.vue')
		},
		{
			path: '*', // Everything that doesn't match something
			name: 'notFound',
			component: () => import(/* webpackChunkName: "not-found" */ './views/NotFound.vue')
		}
	],
	// Native-like scroll behavior
	scrollBehavior(to, from, savedPosition) {
		// If there is an hash, this is an anchor, so scroll to it
		if (to.hash) {
			return {
				selector: to.hash
			};
		} else {
			// Else simulate browser behavior by returning to the previous position, or fallback to top
			if (savedPosition) {
				return savedPosition;
			} else {
				return { x: 0, y: 0 };
			}
		}
	}
});

// Resolve translations before each navigation
router.beforeEach((to, from, next) => {
	const lang = i18n.locale;
	loadLanguageAsync(lang).then(() => next());
});

export default router;
