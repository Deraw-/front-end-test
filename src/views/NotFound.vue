<template>
	<section class="page not-found">
		<h2 class="code">
			404
		</h2>

		<h3>Page non trouvée</h3>

		<p>Il semblerait qu'il y ait eu une erreur !</p>

		<RouterLink
			:to="{ name: 'home' }"
			class="btn primary"
		>
			Retour à la page d'accueil
		</RouterLink>
	</section>
</template>

<script lang="ts">
	import Vue from 'vue';
	import { MetaInfo } from 'vue-meta';

	export default Vue.extend({
		name: 'NotFound',
		metaInfo(): MetaInfo {
			return {
				title: this.$t('views.notFound.meta.title') as string,
				meta: [
					{
						name: 'description',
						content: this.$t('views.notFound.meta.description') as string
					}
				]
			};
		}
	});
</script>

<style lang="scss" scoped>
	.code {
		color: #00a3ba;
		font-size: 6rem;
		font-family: serif;
	}

	p {
		margin-top: 8px;
	}
</style>
