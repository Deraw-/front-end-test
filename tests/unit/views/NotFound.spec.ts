import { shallowMount } from '@vue/test-utils';
import NotFound from '@/views/NotFound.vue';

import localVue from 'tests-unit/index';
import router from 'tests-unit/router';

// Build component before each test
let wrapper: any;
beforeEach(() => {
	wrapper = shallowMount(NotFound, {
		localVue,
		router
	});
});

// Tests
describe('NotFound.vue', () => {
	it('renders correctly', () => {
		expect(wrapper).toMatchSnapshot();
	});
});
