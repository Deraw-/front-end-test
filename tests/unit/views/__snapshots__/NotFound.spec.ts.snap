// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound.vue renders correctly 1`] = `
<section class="page">
  <h2 class="code">
    views.notFound.code
  </h2>
  <h3>views.notFound.title</h3>
  <p>views.notFound.description</p>
  <routerlink-stub to="[object Object]" tag="a" event="click" class="btn primary">
    views.notFound.homeLink
  </routerlink-stub>
</section>
`;
